QuasarHelios2015/src/org/team1540/quasarhelios/AutonomousModeBase.java

```diff
@@ -0,0 +1,55 @@
+package org.team1540.quasarhelios;
+
+import ccre.channel.BooleanInputPoll;
+import ccre.ctrl.BooleanMixing;
+import ccre.instinct.AutonomousModeOverException;
+import ccre.instinct.InstinctModeModule;
+
+public abstract class AutonomousModeBase extends InstinctModeModule {
+	public AutonomousModeBase(String modeName) {
+		super(modeName);
+	}
+	
+	protected void drive(float distance) throws AutonomousModeOverException, InterruptedException {
+		float currentEncoder = DriveCode.leftEncoder.get();
+		DriveCode.allMotors.set(distance > 0 ? 1.0f : -1.0f);
+		
+		if (distance > 0) {
+			waitUntilAtLeast(DriveCode.leftEncoder, currentEncoder + distance);
+		} else {
+			waitUntilAtMost(DriveCode.leftEncoder, currentEncoder + distance);
+		}
+
+		DriveCode.allMotors.set(0.0f);
```

Comment:
Maybe the two comparisons for distance can be combined for clarity. Also, is `currentEncoder` a good name given that it's no longer current during the execution of the `wait...` methods?